common/src/stack/report-system/command/report/system/fixtures/disk_partition_check.py

```diff
@@ -0,0 +1,131 @@
+import pytest
+import paramiko
+import socket
+import re
+from collections import namedtuple
+from itertools import groupby
+
+# Takes in a hostname, returns the lsblk data in a dictionary of disks with a nested
+# dictionary of partitions per disk which each has a named tuple of the name, mountpoint,
+# disk label, file system type, and disk size. Returns an empty dictionary if the host
+# cannot be ssh into or lsblk does not return any data
+def get_partition_data(hostname):
```

Comment:
Actually, you could do the disk grouping and line processing at the same time. So, the `get_partition_data` becomes something like (note: untested):

```
def get_partition_data(hostname):
	disks = {}
	
	backend_ssh = paramiko.SSHClient()
	backend_ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())

	# Skip the test if we can't ssh in
	try:
		backend_ssh.connect(hostname=f'{hostname}')
	
		# Get backend host partition informaiton using lsblk
		stdin, stdout, stderr = backend_ssh.exec_command('lsblk -r -n -o name,mountpoint,label,fstype,size')
		
		for line in sorted(stdout.readlines()):
			parts = line.split()
			if len(parts) == 2:
				disk = parts[0]
				disks[disk] = []
			else:
				disks[disk].append(Partition(*parts))

	except (paramiko.SSHException, socket.error):
		pass

	return disks
```